packages/client/src/v2-events/features/events/AdvancedSearch/utils.ts

```diff
@@ -47,3 +53,126 @@ export const getAdvancedSearchFieldErrors = (
 
 export const flattenFieldErrors = (fieldErrors: Errors) =>
   Object.values(fieldErrors).flatMap((errObj) => errObj.errors)
+
+export const getDefaultSearchFields = (
+  section: AdvancedSearchConfig
+): FieldConfig[] => {
+  const res: FieldConfig[] = []
+  section.fields.forEach((field) => {
+    switch (field.fieldId) {
+      case 'event.trackingId':
+        res.push({
+          id: 'event.trackingId',
+          type: 'TEXT',
+          label: {
+            defaultMessage: 'Tracking ID',
+            description: 'Label for tracking ID field',
+            id: 'v2.advancedSearch.trackingId'
+          }
+        })
+        break
+      case 'event.status':
+        res.push({
+          id: 'event.status',
+          type: 'SELECT',
+          label: {
+            defaultMessage: 'Status of record',
+            description: 'Label for status field',
+            id: 'v2.advancedSearch.status'
+          },
+          options: field.options ?? []
+        })
+        break
+      default:
+        res.push()
+    }
+  })
+
+  return res
+}
+
+const RegStatus = {
+  Archived: 'ARCHIVED',
+  Certified: 'CERTIFIED',
+  CorrectionRequested: 'CORRECTION_REQUESTED',
+  DeclarationUpdated: 'DECLARATION_UPDATED',
+  Declared: 'DECLARED',
+  InProgress: 'IN_PROGRESS',
+  Issued: 'ISSUED',
+  Registered: 'REGISTERED',
+  Rejected: 'REJECTED',
+  Validated: 'VALIDATED',
+  WaitingValidation: 'WAITING_VALIDATION',
+  Created: 'CREATED'
+} as const
+
+type Condition =
+  | { type: 'fuzzy'; term: string }
+  | { type: 'exact'; term: string }
+  | { type: 'range'; gte: string; lte: string }
+  | { type: 'anyOf'; terms: string[] }
+
+export const ADVANCED_SEARCH_KEY = 'and'
+
+function buildCondition(type: string, value: string): Condition {
```

Comment:
We probably want to achieve strict enumeration typing for the `type` here, not just any `string`